### DISCUSSION TITLE
Result.rowsUpdated does not work as expected.

### DISCUSSION THREAD

**Discussion by hantsy** [pgjdbc/r2dbc-postgresql#471] [Q&A] [answered]
I tried to use the following insert statement to get the affected rows, I expected it is 2, but it always return 1 in the `Result.rowsUpdated`.

```kotlin
val sql = "insert into posts(title, content, status) values ($1, $2, $3)";
Mono
.from(template.withTransaction { status: ReactiveTransactionStatus<Connection> ->
    val statement = status.connection.createStatement(sql)
    statement
        .bind(0, "test title")
        .bind(1, "test content")
        .bind(2, "DRAFT")
        .add()
    statement.bind(0, "test2 title")
        .bind(1, "test2 content")
        .bind(2, "DRAFT")
        .add()

    Flux.from(statement.execute()).flatMap { Flux.from(it.rowsUpdated) }
})
.`as` { StepVerifier.create(it) }
.consumeNextWith { it shouldBeEqualComparingTo 1 }
//.consumeNextWith { it shouldBeEqualComparingTo 1 }
.verifyComplete()
```

In the R2dbc spec, it states in [R2dbc batching statement](https://r2dbc.io/spec/0.8.4.RELEASE/spec/html/#statements.batching), it could return one or many `Result` objects that depends on the implementation.

Here it only [returns one `Result`](https://github.com/hantsy/micronaut-sandbox/blob/master/r2dbc-kotlin-co/src/test/kotlin/com/example/PostRepositoryTest.kt#L172), it is OK, but it inserted two records here. In contrast to the Jdbc API, here [it will return an array of the executed results](https://github.com/hantsy/micronaut-sandbox/blob/master/jdbc-kotlin/src/test/kotlin/com/example/PostRepositoryTest.kt#L123).

**Comment by hantsy** [comment, marked as answer]
Oh, it is my fault, should use Flux.from to create the flow.

**Reply by mp911de** [reply]
Using `.add()` enters vendor-specific behavior. Postgres and SQL server issue the statement for each binding set and therefore each statement execution produces one update count element. Databases that accept multiple bindings for one statement could behave differently.
